multiauthenticator/multiauthenticator.py

```diff
@@ -66,6 +67,12 @@ class MultiAuthenticator(Authenticator):
     for JupyterHub"""
 
     authenticators = List(help="The subauthenticators to use", config=True)
+    username_prefix = Unicode(
```

Comment:
Another option (not better/worse, just another option) is to have this as a `Callable(...)` which takes the authenticator as a parameter and returns the prefix, with the default being the current behaviour.

The advantage is flexibility since you could have a mix of shared and unique prefixes, the downside is someone has to define a callable instead of a string if they want a common/no prefix `c.MultiAuthenticator.username_prefix=lambda a: ""`

Reply:
That's an interesting take !

@lahwaacz since you are behind the feature request, I would like to know whether you would prefer @manics suggestion over the current implementation.